pkg/manifests/manifests.go

```diff
@@ -2900,6 +2901,15 @@ func (f *Factory) MonitoringPluginDeployment() (*appsv1.Deployment, error) {
 
 	containers[idx].Image = f.config.Images.MonitoringPlugin
 
+	// Hash the TLS secret and propagate it as an annotation to the
+	// deployment's pods to trigger a new rollout when the TLS certificate/key
+	// are rotated.
+	h := fnv.New64()
+	for _, k := range sets.StringKeySet[[]byte](tlsSecret.Data).List() {
+		h.Write(tlsSecret.Data[k])
```

Comment:
How about just?:

```suggestion
	for _, v := range tlsSecret.Data {
		h.Write(v)
	}
```

Reply:
Some extreme scenarios I was thinking about, we could just ignore them:

I don't know if we should only consider the TLS keys, as we don't want to restart the plugin if anything else gets added to the secret by mistake (ca operator will correct that maybe but what if the change is too frequent/automated?)

Also, if the TLS content get missing or sth (content gets wiped by ca operator bcz of a bug or sth), should we tell the deployment about it or leave it run with the old certificate?

Reply:
I was also thinking the same about the List but didn't think about the hashing

Reply:
selecting only tls.crt and tls.key is more error-prone IMHO but I'll add a comment to clarify.

Reply:
if the ca operator changes that it'll be a breaking change, plus we already count on their stability https://github.com/search?q=repo%3Aopenshift%2Fcluster-monitoring-operator+tls.crt+language%3AGo&type=code&l=Go

But ok, let's keep it simple and see, thanks.